src/Avalonia.Base/Data/Core/ExpressionObserver.cs

```diff
@@ -99,14 +97,14 @@ public class ExpressionObserver : LightweightObservableBase<object?>, IDescripti
         /// </summary>
         /// <param name="rootGetter">A function which gets the root object.</param>
         /// <param name="node">The expression.</param>
-        /// <param name="update">An observable which triggers a re-read of the getter.</param>
+        /// <param name="update">An observable which triggers a re-read of the getter. Generic argument value is not used.</param>
         /// <param name="description">
         /// A description of the expression.
         /// </param>
         public ExpressionObserver(
             Func<object?> rootGetter,
             ExpressionNode node,
-            IObservable<Unit> update,
+            IObservable<object> update,
```

Comment:
`Unit` replaced with `object`.

Reply:
Isn't this class planned to be made internal? We could keep `Unit` type in that case for clarity. It's ridiculous that MS hasn't added that type to BCL yet.

Reply:
It's still planned to be internal. But I don't want to do it in this PR.

Reply:
There's also [`ValueTuple`](https://learn.microsoft.com/en-us/dotnet/api/system.valuetuple?view=net-7.0) (empty tuple).

Reply:
may be DBNull [DBNull](https://learn.microsoft.com/ru-ru/dotnet/api/system.dbnull)

Reply:
Replaced with ValueTuple